projects/packages/search/src/dashboard/class-initial-state.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * The React initial state.
+ *
+ * @package automattic/jetpack-search
+ */
+
+namespace Automattic\Jetpack\Search;
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Automattic\Jetpack\Status;
+use Jetpack_Options;
+
+/**
+ * The React initial state.
+ */
+class Initial_State {
+	/**
+	 * Connection Manager
+	 *
+	 * @var Connection_Manager
+	 */
+	protected $connection_manager;
+
+	/**
+	 * Search Moduel Control
+	 *
+	 * @var Module_Control
+	 */
+	protected $module_control;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Connection_Manager $connection_manager - Connection mananger instance.
+	 * @param Module_Control     $module_control - Module control instance.
+	 */
+	public function __construct( $connection_manager = null, $module_control = null ) {
+		$this->connection_manager = $connection_manager ? $connection_manager : new Connection_Manager( 'jetpack-search' );
+		$this->module_control     = $module_control ? $module_control : new Module_Control();
+	}
+
+	/**
+	 * Render JS for the initial state
+	 *
+	 * @return string - JS string.
+	 */
+	public function render() {
+		return 'var JETPACK_SEARCH_DASHBOARD_INITIAL_STATE=JSON.parse(decodeURIComponent("' . rawurlencode( wp_json_encode( $this->get_initial_state() ) ) . '"));';
+	}
+
+	/**
+	 * Get the initial state data.
+	 *
+	 * @return array
+	 */
+	public function get_initial_state() {
+		return array(
+			'siteData'        => array(
+				'WP_API_root'       => esc_url_raw( rest_url() ),
+				'WP_API_nonce'      => wp_create_nonce( 'wp_rest' ),
+				'registrationNonce' => wp_create_nonce( 'jetpack-registration-nonce' ),
+				'purchaseToken'     => $this->get_purchase_token(),
+				/**
+				 * Whether promotions are visible or not.
+				 *
+				 * @param bool $are_promotions_active Status of promotions visibility. True by default.
+				 */
+				'showPromotions'    => apply_filters( 'jetpack_show_promotions', true ),
+				'adminUrl'          => esc_url( admin_url() ),
+				'blogId'            => Jetpack_Options::get_option( 'id', 0 ),
+				'version'           => defined( 'JETPACK_SEARCH_PACKAGE_VERSION' ) ? JETPACK_SEARCH_PACKAGE_VERSION : 'dev',
```

Comment:
Need a proper way to get version - plugin version or package version?

Reply:
Let's go with the package version so that we don't need to have differing logic for our two plugins.

Reply:
will leave it like this for now. added a TODO here :-)